### DISCUSSION TITLE
Order controller populate method should be able to handle multiple products

### DISCUSSION THREAD

**Discussion by seand7565** [solidusio/solidus#3973] [New Features or Ideas]
I think there are a lot of circumstances where it's desirable for the order controller to be able to accept multiple variants. For instance, if I sell selectable "add-ons" or extras on a product page, that are separate products, I'd want to be able to add both the initial product and the add-on product to the cart at the same time, if it's selected.

I've made this change in other stores, and it's a pretty small one - `order_controller` just needs to be able to accept `params[:variants]` instead of `params[:variant]` and `params[:quantity]`. But it adds a lot of flexibility to the controller, and it's definitely on my wishlist for the platform. What do you all think?

**Comment by jarednorman** [comment]
Do you mean the orders controller, like the populate endpoint?

**Reply by seand7565** [reply]
That's the one. 🤦

**Reply by jarednorman** [reply]
Agreed though. We see a lot of stuff where people have added a multi add that takes variants and quantities.

**Comment by kennyadsl** [comment]
I agree, this is a valid point and seems a low-hanging fruit. Can you open an issue on the project for this?

**Reply by elia** [reply]
Just for reference the issue is this one https://github.com/solidusio/solidus/issues/3977 😊

**Comment by elia** [comment]
Speaking of the populate action… what about also moving [the quantity check](https://github.com/solidusio/solidus/blob/558fa4ba805727060ee661bbe7cfc7a381a1f414/frontend/app/controllers/spree/orders_controller.rb#L58-L61) out of the controller into order-content, and bound it to a preference for the maximum number of items that can be added at once?

**Reply by jarednorman** [reply]
Hmm, a preference would be nice there. A lot of stores aren't going to need it to be more than like 10 or something and could completely shortcut any abuse that could come from that.
